### PR TITLE
improve token errors

### LINKED ISSUE #2: Improve error messages with bad or missing token
When an invalid token is in `.datadelivery.yml`:

```
$ datadelivery deliver -b handover-demo-no-symlinks --email john.bradley@duke.edu
Traceback (most recent call last):
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 41, in _check_response
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/requests-2.18.4-py3.6.egg/requests/models.py", line 935, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: http://127.0.0.1:8000/api/v2/s3-endpoints/?name=default

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/bin/datadelivery", line 11, in <module>
    load_entry_point('datadelivery==0.0.1', 'console_scripts', 'datadelivery')()
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/__main__.py", line 12, in main
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/argparser.py", line 26, in parse_and_run_commands
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/argparser.py", line 73, in _run_deliver
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/commands.py", line 24, in deliver
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/commands.py", line 14, in _create_s3
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 11, in __init__
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 51, in _get_current_endpoint
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 28, in _get_request
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 43, in _check_response
datadelivery.s3.S3Exception: {"detail":"Invalid token."}
```

When no token is in `.datadelivery.yml`:

```
$ datadelivery deliver -b handover-demo-no-symlinks --email john.bradley@duke.edu
Traceback (most recent call last):
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/bin/datadelivery", line 11, in <module>
    load_entry_point('datadelivery==0.0.1', 'console_scripts', 'datadelivery')()
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/__main__.py", line 12, in main
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/argparser.py", line 26, in parse_and_run_commands
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/argparser.py", line 73, in _run_deliver
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/commands.py", line 24, in deliver
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/commands.py", line 13, in _create_s3
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/config.py", line 23, in read_or_create_config
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/config.py", line 48, in read_config
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/config.py", line 59, in __init__
KeyError: 'token'
```

### PR DESCRIPTION
**Pull request by johnbradley**
Makes config file loading more robust and shows detail of remote errors to users.
Also turns on basic circleci automated testing.

Fixes #2

### CODE REVIEW

File: datadelivery/s3.py [hunk 1]
```diff
@@ -40,7 +40,18 @@ def _check_response(response):
...
+        message = response.text
+        try:
+            data = response.json()
+            if 'detail' in data:
+                message = data['detail']
+        except ValueError:
+            pass  # response was not JSON
+        return message
 
     def _get_current_endpoint(self):
```
Comment (dleehr) [inline datadelivery/s3.py:59]: This note is about line 64, still error-handling

I updated my local `.datadelivery.yml` to have a token for the live server. When I try to run a delivery, I get the following error:

```
$ datadelivery deliver -b sample --email first.last@duke.edu
Traceback (most recent call last):
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/bin/datadelivery", line 11, in <module>
    load_entry_point('datadelivery==0.0.1', 'console_scripts', 'datadelivery')()
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/__main__.py", line 14, in main
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/argparser.py", line 26, in parse_and_run_commands
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/argparser.py", line 73, in _run_deliver
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/commands.py", line 24, in deliver
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/commands.py", line 14, in _create_s3
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 11, in __init__
  File "/Users/dcl9/Code/python/datadelivery-cli/venv/lib/python3.6/site-packages/datadelivery-0.0.1-py3.6.egg/datadelivery/s3.py", line 64, in _get_current_endpoint
AttributeError: 'Config' object has no attribute 's3_url'
```

File: datadelivery/config.py [hunk 2]
```diff
@@ -47,18 +48,38 @@ def read_config(self):
...
-            }, stream)
+            yaml.safe_dump(config.to_dict(), stream)
 
 
 class Config(object):
     def __init__(self, data):
-        self.token = data['token']
-        self.url = data.get('url', DEFAULT_DATA_DELIVERY_URL)
-        self.endpoint_name = data.get('endpoint_name', DEFAULT_ENDPOINT_NAME)
+        self.token = data.get('token')
```
Comment (dleehr) [inline datadelivery/config.py:58]: Were these changed to lazily-loaded properties to avoid writing out defaults to the YAML file?

**Review by dleehr — COMMENTED** [review]
Looks good, and thanks for configuring CI. I think we have a typo/missed config property on s3_url and a basic question about why structure changed.

**Comment by dleehr** [comment]
I think we can still improve the prompts and error messages to guide users about what to do, but this is a big step forward for #3

File: datadelivery/config.py (hunk 2 above)
Comment (johnbradley) [reply datadelivery/config.py:58]: Yes.

File: datadelivery/s3.py (hunk 1 above)
Comment (johnbradley) [reply datadelivery/s3.py:59]: I made a change to fix this bug.

**Review by dleehr — APPROVED** [review]

**Pull request merged**